superset/views/alerts.py

```diff
@@ -23,17 +24,62 @@
 from wtforms import BooleanField, Form, StringField
 
 from superset.constants import RouteMethod
-from superset.models.alerts import Alert, AlertLog
+from superset.models.alerts import (
+    Alert,
+    AlertLog,
+    AlertValidatorType,
+    SQLObservation,
+    SQLObserver,
+    Validator,
+)
 from superset.models.schedules import ScheduleType
 from superset.tasks.schedules import schedule_alert_query
+from superset.utils import core as utils
 from superset.utils.core import get_email_address_str, markdown
 
 from ..exceptions import SupersetException
+from ..sql_parse import ParsedQuery
 from .base import SupersetModelView
 
 # TODO: access control rules for this module
 
 
+def test_observer_sql(item: "SQLObserverInlineView") -> None:
```

Comment:
test_ prefix should be used only in the test method / classes

this would be check/validate would be a better name, be aware that validation may take quite some time - but it's nice to have it. validation logic should also live in the task as well, you can abstract this function. It may happen that table schema changed and that would make observer invalid.